### PR TITLE
Comply with Golang semantic import versioning

### PR DESCRIPTION
**Pull request by Loyalsoldier**

### CODE REVIEW

**Comment by Loyalsoldier** [comment]
Article about Semantic Import Versioning: https://research.swtch.com/vgo-import

**Comment by kslr** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ proxy/vless/encoding/addons.pb.go  -3
+ infra/vprotogen/main.go  -17
         

Clones added
============
- proxy/dns/config.pb.go  2
- transport/internet/quic/config.pb.go  3
- proxy/vless/encoding/addons.pb.go  1
- transport/internet/xtls/config.pb.go  1
- transport/internet/headers/tls/config.pb.go  3
- transport/internet/tls/config.pb.go  2
- common/net/port.pb.go  1
- app/log/command/config.pb.go  1
- app/reverse/config.pb.go  1
- transport/internet/tcp/config.pb.go  5
- proxy/vmess/inbound/config.pb.go  2
- transport/internet/headers/srtp/config.pb.go  4
- proxy/vless/account.pb.go  1
- app/log/config.pb.go  1
- app/proxyman/config.pb.go  2
- proxy/trojan/config.pb.go  2
- app/stats/config.pb.go  1
- common/net/address.pb.go  1
- transport/internet/headers/wireguard/config.pb.go  3
- common/protocol/server_spec.pb.go  2
- app/router/config.pb.go  2
- transport/internet/headers/noop/config.pb.go  3
- config.pb.go  1
- common/net/destination.pb.go  1
- transport/internet/headers/utp/config.pb.go  3
- proxy/blackhole/config.pb.go  2
- app/policy/config.pb.go  1
- common/protocol/headers.pb.go  1
- transport/internet/config.pb.go  2
- proxy/dokodemo/config.pb.go  2
- common/log/log.pb.go  1
- transport/internet/websocket/config.pb.go  1
- proxy/mtproto/config.pb.go  2
- app/proxyman/command/command.pb.go  3
- common/net/network.pb.go  1
- common/serial/typed_message.pb.go  1
- proxy/shadowsocks/config.pb.go  2
- transport/internet/domainsocket/config.pb.go  1
- proxy/vmess/outbound/config.pb.go  3
- app/stats/command/command.pb.go  1
- transport/internet/http/config.pb.go  1
- proxy/vless/encoding/encoding_test.go  1
- app/router/command/command.pb.go  2
- app/dns/config.pb.go  2
- app/dispatcher/config.pb.go  1
- transport/config.pb.go  2
- transport/internet/udp/config.pb.go  2
- app/commander/config.pb.go  2
- proxy/http/config.pb.go  2
- proxy/vless/inbound/config.pb.go  2
- common/protocol/user.pb.go  2
- proxy/vless/outbound/config.pb.go  3
- proxy/vmess/account.pb.go  2
- transport/internet/kcp/config.pb.go  4
- proxy/vmess/encoding/encoding_test.go  1
- proxy/socks/config.pb.go  2
- transport/internet/headers/http/config.pb.go  4
- proxy/freedom/config.pb.go  2
- transport/internet/headers/wechat/config.pb.go  3
         

Clones removed
==============
+ infra/vprotogen/main.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/v2fly/v2ray-core/pullRequest?prid=6416992&bid=20790781)

**Comment by kslr** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ infra/vprotogen/main.go  -17
+ proxy/vless/encoding/addons.pb.go  -3
         

Clones added
============
- app/log/command/config.pb.go  1
- proxy/vless/outbound/config.pb.go  3
- proxy/socks/config.pb.go  2
- proxy/blackhole/config.pb.go  2
- transport/internet/domainsocket/config.pb.go  1
- transport/internet/headers/utp/config.pb.go  3
- common/serial/typed_message.pb.go  1
- app/proxyman/config.pb.go  2
- proxy/http/config.pb.go  2
- common/protocol/server_spec.pb.go  2
- proxy/freedom/config.pb.go  2
- app/stats/config.pb.go  1
- proxy/vmess/account.pb.go  2
- proxy/mtproto/config.pb.go  2
- common/protocol/user.pb.go  2
- common/net/address.pb.go  1
- transport/config.pb.go  2
- proxy/trojan/config.pb.go  2
- transport/internet/config.pb.go  2
- proxy/vless/account.pb.go  1
- proxy/shadowsocks/config.pb.go  2
- app/dns/config.pb.go  2
- transport/internet/xtls/config.pb.go  1
- transport/internet/tcp/config.pb.go  5
- transport/internet/headers/wechat/config.pb.go  3
- transport/internet/quic/config.pb.go  3
- common/net/destination.pb.go  1
- proxy/dns/config.pb.go  2
- app/commander/config.pb.go  2
- config.pb.go  1
- common/log/log.pb.go  1
- proxy/vmess/encoding/encoding_test.go  1
- transport/internet/headers/srtp/config.pb.go  4
- proxy/vmess/inbound/config.pb.go  2
- app/router/config.pb.go  2
- proxy/vless/encoding/encoding_test.go  1
- app/proxyman/command/command.pb.go  3
- transport/internet/headers/wireguard/config.pb.go  3
- proxy/dokodemo/config.pb.go  2
- app/dispatcher/config.pb.go  1
- transport/internet/headers/noop/config.pb.go  3
- transport/internet/headers/http/config.pb.go  4
- app/policy/config.pb.go  1
- common/net/network.pb.go  1
- app/log/config.pb.go  1
- common/net/port.pb.go  1
- proxy/vless/inbound/config.pb.go  2
- app/reverse/config.pb.go  1
- proxy/vless/encoding/addons.pb.go  1
- app/router/command/command.pb.go  2
- transport/internet/kcp/config.pb.go  4
- transport/internet/tls/config.pb.go  2
- transport/internet/http/config.pb.go  1
- common/protocol/headers.pb.go  1
- app/stats/command/command.pb.go  1
- transport/internet/headers/tls/config.pb.go  3
- transport/internet/udp/config.pb.go  2
- proxy/vmess/outbound/config.pb.go  3
- transport/internet/websocket/config.pb.go  1
         

Clones removed
==============
+ infra/vprotogen/main.go  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/v2fly/v2ray-core/pullRequest?prid=6416992&bid=20790781)

**Comment by kslr** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- transport/config.pb.go  2
- app/stats/config.pb.go  1
- transport/internet/quic/config.pb.go  3
- common/net/address.pb.go  1
- transport/internet/config.pb.go  2
- app/router/command/command.pb.go  2
- proxy/vmess/outbound/config.pb.go  3
- app/proxyman/command/command.pb.go  3
- proxy/vmess/encoding/encoding_test.go  1
- proxy/vless/inbound/config.pb.go  2
- transport/internet/kcp/config.pb.go  4
- common/log/log.pb.go  1
- app/log/config.pb.go  1
- app/dns/config.pb.go  2
- common/net/network.pb.go  1
- proxy/shadowsocks/config.pb.go  2
- proxy/trojan/config.pb.go  2
- common/protocol/user.pb.go  2
- transport/internet/tls/config.pb.go  2
- proxy/http/config.pb.go  2
- app/proxyman/config.pb.go  2
- proxy/freedom/config.pb.go  2
- transport/internet/websocket/config.pb.go  1
- transport/internet/headers/wechat/config.pb.go  3
- proxy/vmess/account.pb.go  2
- app/router/config.pb.go  2
- proxy/vless/outbound/config.pb.go  3
- transport/internet/headers/srtp/config.pb.go  4
- common/serial/typed_message.pb.go  1
- proxy/socks/config.pb.go  2
- common/protocol/headers.pb.go  1
- app/stats/command/command.pb.go  1
- app/policy/config.pb.go  1
- transport/internet/headers/wireguard/config.pb.go  3
- app/log/command/config.pb.go  1
- transport/internet/domainsocket/config.pb.go  1
- common/net/port.pb.go  1
- transport/internet/tcp/config.pb.go  5
- config.pb.go  1
- transport/internet/headers/utp/config.pb.go  3
- common/net/destination.pb.go  1
- app/dispatcher/config.pb.go  1
- transport/internet/udp/config.pb.go  2
- transport/internet/xtls/config.pb.go  1
- app/commander/config.pb.go  2
- proxy/dns/config.pb.go  2
- transport/internet/headers/http/config.pb.go  4
- app/reverse/config.pb.go  1
- proxy/vless/encoding/encoding_test.go  1
- proxy/blackhole/config.pb.go  2
- common/protocol/server_spec.pb.go  2
- transport/internet/headers/tls/config.pb.go  3
- proxy/vless/account.pb.go  1
- transport/internet/headers/noop/config.pb.go  3
- proxy/vmess/inbound/config.pb.go  2
- proxy/dokodemo/config.pb.go  2
- proxy/mtproto/config.pb.go  2
- transport/internet/http/config.pb.go  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/v2fly/v2ray-core/pullRequest?prid=6416992&bid=20790781)

File: app/proxyman/command/command.pb.go [hunk 1]
```diff
@@ -8,13 +8,13 @@ package command
 
 import (
 	proto "github.com/golang/protobuf/proto"
+	v5 "github.com/v2fly/v2ray-core/v5"
```
Comment (kslr) [inline app/proxyman/command/command.pb.go:11]: 我看其他的部分写的是 core 可能统一下比较好

File: app/commander/config.proto [hunk 2]
```diff
@@ -2,7 +2,7 @@ syntax = "proto3";
 
 package v2ray.core.app.commander;
 option csharp_namespace = "V2Ray.Core.App.Commander";
-option go_package = "v2ray.com/core/app/commander";
+option go_package = "github.com/v2fly/v2ray-core/v5/app/commander";
```
Comment (kslr) [inline app/commander/config.proto:5]: 这里的改动会不会影响 grpc 呢

**Review by kslr — COMMENTED** [review]
先提交一部分

File: app/proxyman/command/command.pb.go (hunk 1 above)
Comment (Loyalsoldier) [reply app/proxyman/command/command.pb.go:11]: 这是生成的代码

File: app/commander/config.proto (hunk 2 above)
Comment (Loyalsoldier) [reply app/commander/config.proto:5]: gPRC 跟包名耦合吗？不太清楚。但是能生成 pb.go 文件，并测试通过应该没啥问题吧

**Comment by kslr** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- proxy/vless/encoding/encoding_test.go  1
- common/serial/typed_message.pb.go  1
- app/proxyman/config.pb.go  2
- transport/internet/tcp/config.pb.go  5
- common/net/port.pb.go  1
- proxy/vless/account.pb.go  1
- transport/internet/websocket/config.pb.go  1
- proxy/blackhole/config.pb.go  2
- app/dns/config.pb.go  2
- common/log/log.pb.go  1
- app/stats/command/command.pb.go  1
- proxy/vmess/account.pb.go  2
- transport/internet/headers/tls/config.pb.go  3
- app/commander/config.pb.go  2
- common/protocol/server_spec.pb.go  2
- proxy/trojan/config.pb.go  2
- transport/internet/http/config.pb.go  1
- transport/internet/xtls/config.pb.go  1
- proxy/dokodemo/config.pb.go  2
- config.pb.go  1
- transport/internet/udp/config.pb.go  2
- common/net/destination.pb.go  1
- transport/internet/quic/config.pb.go  3
- app/dispatcher/config.pb.go  1
- proxy/http/config.pb.go  2
- proxy/vless/inbound/config.pb.go  2
- common/protocol/headers.pb.go  1
- app/router/command/command.pb.go  2
- proxy/socks/config.pb.go  2
- proxy/dns/config.pb.go  2
- proxy/shadowsocks/config.pb.go  2
- transport/internet/tls/config.pb.go  2
- proxy/vmess/inbound/config.pb.go  2
- app/log/config.pb.go  1
- app/stats/config.pb.go  1
- transport/internet/headers/utp/config.pb.go  3
- transport/internet/config.pb.go  2
- proxy/freedom/config.pb.go  2
- app/log/command/config.pb.go  1
- transport/internet/kcp/config.pb.go  4
- proxy/vless/outbound/config.pb.go  3
- proxy/vmess/outbound/config.pb.go  3
- common/net/network.pb.go  1
- transport/config.pb.go  2
- transport/internet/domainsocket/config.pb.go  1
- common/protocol/user.pb.go  2
- app/reverse/config.pb.go  1
- common/net/address.pb.go  1
- proxy/vmess/encoding/encoding_test.go  1
- transport/internet/headers/noop/config.pb.go  3
- transport/internet/headers/http/config.pb.go  4
- transport/internet/headers/srtp/config.pb.go  4
- proxy/mtproto/config.pb.go  2
- app/proxyman/command/command.pb.go  3
- app/policy/config.pb.go  1
- app/router/config.pb.go  2
- transport/internet/headers/wechat/config.pb.go  3
- transport/internet/headers/wireguard/config.pb.go  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/v2fly/v2ray-core/pullRequest?prid=6416992&bid=20790781)

**Comment by ququ0755** [comment]
what do you do this?

**Review by kslr — APPROVED** [review]

**Comment by Loyalsoldier** [comment]
Ready to be merged.

**Comment by Loyalsoldier** [comment]
Leave this step to the last. Or it will be conflict all the way. So close it now.

**Pull request closed without merging**